pkg/client/restclient/request.go

```diff
@@ -811,7 +813,14 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
 		}
 		if err != nil {
-			return err
+			operr, ok := err.(*net.OpError)
+			if ok && operr.Err.Error() == syscall.ECONNRESET.Error() {
```

Comment:
Creating the utility function would also allow you to invert this condition and just have `if !utilnet.IsConnectioReset(err) { return err }`